### ISSUE TITLE
Could be nice to generate/install pkg-config file

### ISSUE THREAD

**Issue by Lecrapouille** [vasild/cpp-ipfs-http-client#63]
I'm testing your project right now.

Could be nice to create a pkg-config file like in my Makefile:
https://github.com/Lecrapouille/MyMakefile/blob/master/Makefile.macros#L165

sudo make install will install it:
https://github.com/Lecrapouille/MyMakefile/blob/master/Makefile.macros#L181
https://github.com/Lecrapouille/MyMakefile/blob/master/Makefile.macros#L63
(not sure this is working well for Mac)

So in your readme instead of:
```
g++ -std=c++11 -I/path/to/header -L/path/to/lib -lipfs-http-client myprog.cc -o myprog
```
Could be replaced by:
```
g++ -std=c++11 myprog.cc -o myprog `pkg-config --cflags --libs ipfs-http-client`
```

**Comment by melroy89** [comment]
I see, thank you for you remarks. Ideally you should use CMake, which interact very well with an existing CMake project.

But I will test if `pkg-config` config you provided also work, instead of the manual `-I ..` and `-L ..`, which is indeed old school.

**Comment by Lecrapouille** [comment]
pkg-config generation is managed by CMakefile. I'll check if I can make a PR from this project I helped, I'm noob with CMake:

https://github.com/sebastiandev/zipper/blob/master/zipper.pc.cmakein

https://github.com/sebastiandev/zipper/blob/87b14a4b31771ed94ebb3204dc6ed69ad9c78436/CMakeLists.txt#L465

https://github.com/sebastiandev/zipper/blob/87b14a4b31771ed94ebb3204dc6ed69ad9c78436/CMakeLists.txt#L479

**Comment by melroy89** [comment]
Yes so `zipper.pc.cmakein` is an input file for the `configure_file` of CMake, generating the final file. Which is put here: `${PROJECT_BINARY_DIR}/zipper.pc`, so under the CMAKE [project binary directory](https://cmake.org/cmake/help/latest/variable/PROJECT_BINARY_DIR.html) with final name 'zipper.pc'. `Configure_file` will fill all the `@CMAKE_INSTALL_PREFIX@` statement etc.
